appinventor/components/src/com/google/appinventor/components/runtime/Navigation.java

```diff
@@ -0,0 +1,352 @@
+  private final Activity activity;
+  private String apiKey;
+  private GeoPoint startLocation;
+  private GeoPoint endLocation;
+  private TransportMethod method;
+  private Web web;
+
+  enum TransportMethod {
+    DEFAULT ("foot-walking"),
+    DRIVING ("driving-car"),
+    CYCLING ("cycling-regular"),
+    WALKING ("foot-walking"),
+    WHEELCHAIR ("wheelchair");
+
+    private final String method;
+    
+    TransportMethod(String method) {
+      this.method = method;
+    }
+    
+    private String method() { 
+      return method;
+    }
+  }
+
+  /**
+   * Creates a Navigation component.
+   *
+   * @param container container, component will be placed in
+   */
+  public Navigation(ComponentContainer container) {
+    super(container.$form());
+    activity = container.$context();
+    apiKey = "";
+    startLocation = new GeoPoint(0, 0);
+    endLocation = new GeoPoint(0, 0);
+    method = TransportMethod.DEFAULT;
+  }
+
+  @SimpleFunction(description = "")
+  public void RequestDirections() {
+    AsynchUtil.runAsynchronously(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          performRequest();
+        } catch (IOException e) {
+          form.dispatchErrorOccurredEvent(Navigation.this, "RequestDirections",
+              ErrorMessages.ERROR_DEFAULT);
+        } catch (JSONException je) {
+          form.dispatchErrorOccurredEvent(Navigation.this, "RequestDirections",
+              ErrorMessages.ERROR_DEFAULT);
+        }
+      }
+    });
+  }
+
+  private void performRequest() throws IOException, JSONException {
+    final String finalURL = String.format(OPEN_ROUTE_SERVICE_URL, this.method.method());
+    URL url = new URL(finalURL);
+    HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+    connection.setDoInput(true);
+    connection.setDoOutput(true);
+    connection.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
+    connection.setRequestMethod("POST");
+    connection.setRequestProperty("Authorization", apiKey);
+    if (connection != null) {
+      try {
+        String coords = "{\"coordinates\": " + JsonUtil.getJsonRepresentation(getCoordinates()) + "}";
+        byte[] postData = coords.getBytes(StandardCharsets.UTF_8);
+        connection.setFixedLengthStreamingMode(postData.length);
+        BufferedOutputStream out = new BufferedOutputStream(connection.getOutputStream());
+        try {
+          out.write(postData, 0, postData.length);
+          out.flush();
+        } finally {
+          out.close();
+        }
+        
+        final String geoJson = getResponseContent(connection);
+        Log.d(TAG, geoJson);
+        final String coordinates = getCoordinatesFromGeoJson(geoJson);
+
+        activity.runOnUiThread(new Runnable() {
+          @Override
+          public void run() {
+            GotDirections(coordinates);
+          }
+        });
+      } catch (Exception e) {
+        e.printStackTrace();
```

Comment:
Log this using the android logging facility and consider reporting an error to the blocks by either using Form's dispatchErrorOccurredEvent method or defining an error event specific to this component (or both, use the specific event handler and fall back to Form if EventDispatcher.dispatchEvent returns false).